game/hud/src/components/AnchorIndicator/index.tsx

```diff
@@ -0,0 +1,72 @@
+
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as React from 'react';
+import {Position} from '../../services/session/layout';
+
+export interface AnchorIndicatorProps {
+  name: string;     // widget name
+  pos: Position;    // current widget position
+  setAnchor: (name: string, pos: Position) => void;
+}
+
+export interface AnchorIndicatorState {
+}
+
+class AnchorIndicator extends React.Component<AnchorIndicatorProps, AnchorIndicatorState> {
+
+  constructor(props: AnchorIndicatorProps) {
+    super(props);
+  }
+
+  private anchorHere = (event: React.MouseEvent) => {
+    const el: HTMLDivElement = event.currentTarget as HTMLDivElement;
+    function a2n(a: string) : number {
+      switch(a) {
+        case 's': return -1;
+        case 'e': return 1;
+      }
+      return 0;
+    }
+    const pos: Position = Object.assign({}, this.props.pos);
+    let x: number = a2n(el.classList[1][7]);
+    let y: number = a2n(el.classList[2][7]);
```

Comment:
Your analysis is correct.  

As this component controls the class names of this element it can be fairly certain that the assumptions on positions are correct, unless something external to this component starts fiddling with classes, which would be bad etiquette.  In fact as a black box, it can insist that you don't tamper with classes.

The alternative would be to add an additional data attribute to the element, or to loop across class names looking for the start/end classes, but I decided it was unnecessary.

As this PR is unlikely to be merged because JB is working on a new system that makes this obsolete, I probably won't be fixing it.